presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -408,6 +409,16 @@ public PlanNode visitTableWriter(TableWriterNode node, RewriteContext<Context> c
             return context.defaultRewrite(node, context.get());
         }
 
+        @Override
+        public PlanNode visitDelete(DeleteNode node, RewriteContext<Context> context)
+        {
+            if (node.getInputDistribution().isPresent()) {
+                context.get().variables.addAll(node.getInputDistribution().get().getInputVariables());
+            }
+            context.get().variables.add(node.getRowId());
+            return context.defaultRewrite(node, context.get());
```

Comment:
@jaystarshot  can you review again? I just changed this line from `return super.visitDelete(node, context);` to `context.defaultRewrite(node, context.get())` like other functions in this class. Thanks!